### DISCUSSION TITLE
How can I deploy rook-ceph with RDMA ?

### DISCUSSION THREAD

**Discussion by mc956759** [rook/rook#12723] [Q&A]
I want to try ceph rdma features. Can I deploy ceph with RDMA by rook?
Any help will be appricated. 
Thanks a lot!

**Comment by travisn** [comment]
This could be enabled with a [Ceph config override](https://rook.io/docs/rook/latest/Storage-Configuration/Advanced/ceph-configuration/#custom-cephconf-settings), I believe you would need to set the [ms_type](https://docs.ceph.com/en/latest/rados/configuration/network-config-ref/#confval-ms_type):
```
[global]
ms_type = async+rdma
```

**Reply by mc956759** [reply]
Is it a prerequisite to install the rdma plug-in of k8s before modifying the configuration? 
I tried to modify ms_type = async+rdma, and restart the osd-pod. it's stuck at initialization:
rook-ceph      rook-ceph-osd-0-86dbb97dc5-vmj22                     0/2     Init:0/2           0                 83s
And describe pod shows:
![image](https://github.com/rook/rook/assets/115455187/10a26800-ee4c-40a8-895d-df82f4842bc8)

**Reply by mc956759** [reply]
I have deployed rdma-shared-device-plugin on k8s, and tested ib_write_bw between diff pods.
But the ceph-osd pod is still stuck at initialization. Should I build osd docker image with rdma related libs?

**Reply by mc956759** [reply]
It seems like infiniband can't init.
![image](https://github.com/rook/rook/assets/115455187/fc192371-8b7a-4fa6-9d94-59906922c453)
![image](https://github.com/rook/rook/assets/115455187/27261da7-9919-42ee-8ceb-67b9fd3cc540)

**Reply by travisn** [reply]
@rzarzynski Ideas on what might be missing to start the cluster with rdma enabled?

**Reply by rzarzynski** [reply]
@travisn unfortunately no but hasn't touched this area at all. However, we've gotten folks in the NVMe-oF project.
